orderer/common/channelparticipation/restapi.go

```diff
@@ -9,21 +9,35 @@ package channelparticipation
 import (
 	"encoding/json"
 	"net/http"
+	"path"
+	"strings"
 
+	"github.com/gorilla/mux"
+	"github.com/hyperledger/fabric/common/configtx"
 	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/hyperledger/fabric/orderer/common/localconfig"
+	"github.com/hyperledger/fabric/orderer/common/types"
 	"github.com/pkg/errors"
 )
 
-const URLBaseV1 = "/participation/v1/"
-
-type ErrorResponse struct {
-	Error string `json:"error"`
-}
+const (
+	URLBaseV1           = "/participation/v1/"
+	URLBaseV1Channels   = URLBaseV1 + "channels"
+	channelIDKey        = "channelID"
+	urlWithChannelIDKey = URLBaseV1Channels + "/{" + channelIDKey + "}"
+)
 
 //go:generate counterfeiter -o mocks/channel_management.go -fake-name ChannelManagement . ChannelManagement
 
 type ChannelManagement interface {
+	// ListAll returns the names of all channels (including the system channel), and the name of the system channel
+	// (empty if does not exist). The URL field is empty, and is to be completed by the caller.
+	ListAllChannels() *types.ChannelList
```

Comment:
The more idiomatic name is just **Channels()**. 
I also don't understand why the channel list is a pointer type. 
Under what circumstance do we return nil? Why not return a concrete type?

Reply:
The pointer type is because the struct can be big, having all those channel names in there, so this way a copy is avoided.

The names corresponds to the API:

List *          -> ListAllChannels()
List x          -> ListChannel(x)
Join x         -> JoinChannel(x)
Remove x  -> RemoveChannel(x)

That is the plan, at least.

Reply:
But the Channels slice inside the struct, is only initialized once, and its elements are not going to be copied even if you use a non pointer type. 

Look at this [go playground sample](https://play.golang.org/p/Cn_BPDlIJm2).
Copying a slice is cheap because you only copy 2 pointers and not the elements inside

Reply:
You are correct. I'll change that.